crypto/src/lib.rs

```diff
@@ -16,6 +16,19 @@
 //!   [RandomCoin] implementation uses a cryptographic hash function to generate pseudo-random
 //!   elements form a seed.
 
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(not(any(feature = "std", feature = "alloc")))]
+compile_error!("Either feature \"std\" or \"alloc\" must be enabled for this crate.");
+
+#[cfg(feature = "alloc")]
+#[macro_use]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+#[macro_use]
+extern crate std;
```

Comment:
I'm curious why do we need this here (and in other crates). It seems to compile fine without these lines.

Reply:
That's an old habit I have 🙃 I think it would have been needed pre Rust 2018 but `extern crate std` is already implicit I believe if `std` is enabled that's right. I'll remove them.

Reply:
Actually (the above answer is only partially correct), we can remove it because the no_std requirement is only enabled conditionally through `#![cfg_attr(not(feature = "std"), no_std)]`

If we had instead `#![no_std]`, we would be forced to include the `extern crate std` conditionally, as otherwise we could not use the standard functions. As here we only conditionally require `no_std`, indeed that can be removed.

Reply:
I'll keep the import and change the crate roots to #![no_std], I believe it's more common among no-std crates (I'm definitely not a specialist on the question 😅)

Reply:
Sorry for the several force-push, changing to `#![no_std]` in the crate roots required to add some imports from utils::collections in test modules. Should be all good now